refactored contract functions out into sep file

W3stside opened the pull request:
Refactored out contract code into sep file

W3stside committed:
refactored contract functions out into sep file

josojo commented:
I think the fact that the third test command is not going through has nothing to do with your specific implementation. Had these issues before. Maybe accounts ran out of funding. Idk.
  
I like the refactoring a LOT

Velenir commented on test/testFunctions.js:
```diff
@@ -0,0 +1,171 @@
+ * deploys all contracts in build/contracts
+ */
+const getContracts = async () => {
+  const depContracts = contractNames.map(c => artifacts.require(c)).map(async cc => cc.deployed())
```
`async` is unnecessary here

W3stside replied:
yep no idea why thats in there O.O

Velenir commented on test/testFunctions.js:
```diff
@@ -0,0 +1,171 @@
+    eth.deposit({ from: acct, value: 10 ** 9 })
+    eth.approve(dx.address, 10 ** 9, { from: acct })
+    gno.transfer(acct, 10 ** 18, { from: accounts[0] })
+    gno.approve(dx.address, 10 ** 18, { from: acct })
```
`await Promise.all` on line 38 doesn't ultimately wait for the `eth.* gno.*` transactions, but only starts them. `accounts.map` maps to `Array<undefined>` not promises. Thanks to the sequential nature of `testrpc`'s transaction handling, it won't usually break anything, but the right thing would be to wrap them in `Promise.all()`:

```
return Promise.all([eth.deposit, eth.approve, gno.transfer, gno.approve])
```

Same with lines 51, 52 below.

W3stside replied:
true true

W3stside committed:
removed unnecessary async

W3stside committed:
linting

W3stside committed:
fixed most BULLSHIT bug ever concerning the INCORRECT auctionStart fn
--> dx.auctionStart SHOULD HAVE BEEN dx.getAuctionStart
--> Tests are PASSING x 4 ... yolo

W3stside committed:
Merge branch 'contracts/feature/NewSolidityCodeFromDom' into contracts/feature/David/RefactoringTestFunctions
